packages/itwin/map-layers/src/ui/FeatureInfoUiItemsProvider.tsx

```diff
@@ -32,14 +37,70 @@ export const getMapFeatureInfoToolItemDef = (): ToolItemDef =>
       await IModelApp.tools.run(MapFeatureInfoTool.toolId);
     },
     badgeType: BadgeType.TechnicalPreview,
+    isHidden: new ConditionalBooleanValue(() => {
+      // Hide the MapFeatureInfoTool if the Map Layers toggle is off or no ArcGISFeature layers are active
+      let isHidden = true;
+      if (!FeatureInfoUiItemsProvider.mapLayerProviders) {
+        FeatureInfoUiItemsProvider.loadMapLayerProviders();
+      }
+      FeatureInfoUiItemsProvider.mapLayerProviders?.forEach((provider) => {
+        if (provider?.supportsMapFeatureInfo) {
+          isHidden = false;
+          return;
+        }
+      });
+      return isHidden;
+    }, [MapLayersSyncUiEventId.MapImageryChanged, SyncUiEventId.ActiveViewportChanged, SyncUiEventId.ViewStateChanged]),
   });
 
 export class FeatureInfoUiItemsProvider implements UiItemsProvider {
   // eslint-disable-line deprecation/deprecation
   public readonly id = "FeatureInfoUiItemsProvider";
   public static readonly widgetId = "map-layers:mapFeatureInfoWidget";
+  private static _mapLayerProviders: MapLayerImageryProvider[] | undefined = undefined;
+
+  public constructor(private _featureInfoOpts: MapFeatureInfoOptions) {
+    IModelApp.viewManager.onViewOpen.addOnce((vp: ScreenViewport) => {
+      const handleMapImageryChanged = (_args: Readonly<MapImagerySettings>) => {
+        FeatureInfoUiItemsProvider.loadMapLayerProviders();
+      };
+      const handleDisplayStyleChanged = () => {
+        FeatureInfoUiItemsProvider.loadMapLayerProviders();
+      };
+      vp.displayStyle.settings.onMapImageryChanged.addListener(handleMapImageryChanged);
```

Comment:
We never unregister those handlers, are we sure this is ok?

Reply:
Resolved by eliminating handlers